### DISCUSSION TITLE
Java version support

### DISCUSSION THREAD

**Discussion by kekbur** [quarkusio/quarkus#36068] [Community]
Does Quarkus have its supported Java versions documented anywhere? I can find blog posts that say 11 is the minimum and 17 is the recommended version, but it would be nice to have some obvious and up-to-date place for this information.

~~The reason I ask is that I was eager to test Quarkus with OpenJDK 21 released yesterday, and found a problem with the way `List` query parameters are handled by Resteasy Reactive client. The problem does not appear with Java 17 or 20.~~

~~Given this API client:~~
```java
@ApplicationScoped
@RegisterRestClient(configKey = "test-api", baseUri = "https://httpbin.org")
public interface TestApi
{
	@GET
	@Path("/get")
	public Map<String, Object> getRequest(@QueryParam("name") List<String> values);
}
```

~~This test fails under Java 21:~~
```java
Map<String, Object> response = api.getRequest(List.of("one", "two"));
assertEquals("https://httpbin.org/get?name=one&name=two", response.get("url"));
```

~~The resulting incorrect request URL is this: `https://httpbin.org/get?name=[one, two]`~~

The bug has been fixed: https://github.com/quarkusio/quarkus/issues/36170

**Comment by brunobat** [comment]
The recommended Java versions can be seen in the github repo at https://github.com/quarkusio/quarkus:
<img width="653" alt="Screenshot 2023-10-18 at 09 31 13" src="https://github.com/quarkusio/quarkus/assets/2318030/054bf5db-fbba-48ab-bc7f-f94569b0a677">

**Reply by kekbur** [reply]
Oh, I missed that. Thank you for pointing it out.
